File: codes/3.ProgrammingLanguage/编程语言试水/Score100WordsAnalyzer.py
import time

start = time.time()
score100_words = []
with open("Dictionary.txt", "r", encoding="utf-8") as file:
    for line in file:
        if not any([char in line for char in ["-", " ", "="]]):
            if sum(ord(c) - ord('a') + 1 for c in line.strip()) == 100:
                score100_words.append(line.strip())

print(score100_words)
print("\nWords number:", len(score100_words))
print("        time: {}ms".format(round((time.time() - start) * 1000, 1)))

# 编码体验
# 1、没有分号
# 2、没有花括号，要用缩进来表示代码块
# 3、没有 let 和 var 关键字，直接写变量名就可以了
# 4、没有类型声明，也没有类型检查
# 5、没有 main 函数
# 6、代码层级太多了，没有链式编程

File: codes/3.ProgrammingLanguage/编程语言试水/Score100WordsAnalyzer.R
start <- Sys.time()
lines <- readLines("Dictionary.txt", encoding = "UTF-8")
lines <- lines[grepl("^[a-z]+$", lines)]
score100Words <- NULL
a <- as.numeric(charToRaw("a"))
for (line in lines) {
  charArray <- strsplit(line, "")[[1]]
  sum <- 0
  for (c in charArray) {
    sum <- sum + as.numeric(charToRaw(c)) - a + 1
  }
  if (sum == 100) {
    score100Words <- c(score100Words, line)
  }
}
print(score100Words)
print(paste("words number:", length(score100Words)))
print(paste("        time:", round((Sys.time() - start) * 1000, 0), "ms"))
